integration/spark/src/test/common/java/io/openlineage/spark/agent/lifecycle/MatchesMapRecursively.java

```diff
@@ -34,25 +37,54 @@ public MatchesMapRecursively(Map<String, Object> target) {
 
   public static Predicate<Map<String, Object>> predicate(
       Map<String, Object> target, Set<String> ommittedKeys) {
+    log.error("??? {}", target);
     return (map) -> {
+      log.error("NEW_MAP: {} - {}", map, target);
       if (!map.keySet().containsAll(target.keySet())) {
+        log.error("KS: {} {}", map.keySet(), target.keySet());
         return false;
       }
       for (String k : target.keySet()) {
-        if (!ommittedKeys.contains(k)) {
+        log.error("KEYSET {}", k);
+        if (ommittedKeys.contains(k)) {
           continue;
         }
         Object val = map.get(k);
         boolean eq;
+        log.error("START_COMPARE");
         if (val instanceof Map) {
           eq =
               MatchesMapRecursively.predicate((Map<String, Object>) target.get(k), ommittedKeys)
                   .test((Map<String, Object>) val);
+        } else if (val instanceof List) {
+          List l = (List) val;
+          List r = (List) target.get(k);
+          if (r.size() != l.size()) {
+            return false;
+          }
+          for (int i = 0; i < l.size(); i++) {
+            if (l.get(i) instanceof Map) {
+              if (MatchesMapRecursively.predicate((Map<String, Object>) r.get(i), ommittedKeys)
+                      .test((Map<String, Object>) l.get(i))
+                  == false) {
```

Comment:
Can we use `!` instead of having `== false` on its own line?

Reply:
Done,